packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2232,7 +2239,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       state: N.ParseClassMemberState,
       isStatic: boolean,
     ): void {
-      this.tsParseModifiers(member, ["abstract", "readonly", "declare"]);
+      this.tsParseModifiers(member, [
+        "abstract",
+        "readonly",
+        "declare",
+        "override",
+      ]);
 
       const idx = this.tsTryParseIndexSignature(member);
```

Comment:
Can you add a new test case?
```ts
class C extends class {} {
  override [key: string]: string;
}
```
TS Nightly will throw
> 'override' modifier cannot appear on an index signature.

Reply:
Going to close this in favor of https://github.com/babel/babel/pull/13097, which handles this case.